### PR TITLE
[FEATURE] Add tests for `SimpleCheckpoint` utilizing Fluent Datasources with Pandas, Spark, and SQLAlchemy test cases.

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Add tests for `SimpleCheckpoint` utilizing style Fluent Datasources with Pandas, Spark, and SQLAlchemy test cases..
* Clean up existing `Checkpoint` tests.

### Next TODO
* Add integration test for `Checkpoint` and `SimpleCheckpoint` utilizing Spark style Fluent Datasource specifying `Schema` for DataAsset (when adding `DataAsset`).
* Add integration test for `Checkpoint` and `SimpleCheckpoint` utilizing Pandas and Spark style Fluent Datasource specifying `FilePathDataAsset` (e.g., using local filesystem for `DataAsset` test case examples).

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], or [CONTRIB]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: DX-469/DX-470
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in GitHub issues or Slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/checkpoint/test_checkpoint_with_fluent_datasources.py [hunk 1]
```diff
@@ -1099,10 +1053,10 @@ def test_newstyle_checkpoint_instantiates_and_produces_a_validation_result_when_
     # create expectation suite
     context.add_expectation_suite("my_expectation_suite")
 
-    batch_request: FluentBatchRequest = FluentBatchRequest(
-        datasource_name="my_sqlite_datasource",
-        data_asset_name="table_partitioned_by_date_column__A_query_asset_limit_5",
-    )
+    batch_request = {
```
Comment (alexsherstinsky) [inline tests/checkpoint/test_checkpoint_with_fluent_datasources.py:1058]: This is the correct change, due to the type requirements in the `Checkpoint` configuration.  There is a JIRA ticket to re-examine the `Checkpoint` design, including proposing emphasis on typed objects and formal interfaces.

File: tests/checkpoint/test_checkpoint_with_fluent_datasources.py [hunk 2]
```diff
@@ -1196,15 +1150,15 @@ def test_newstyle_checkpoint_instantiates_and_produces_a_validation_result_when_
     # create expectation suite
     context.add_expectation_suite("my_expectation_suite")
 
-    batch_request_0: FluentBatchRequest = FluentBatchRequest(
-        datasource_name="my_sqlite_datasource",
-        data_asset_name="table_partitioned_by_date_column__A_query_asset_limit_5",
-    )
+    batch_request_0 = {
```
Comment (alexsherstinsky) [inline tests/checkpoint/test_checkpoint_with_fluent_datasources.py:1155]: This is the correct change, due to the type requirements in the `Checkpoint` configuration.  There is a JIRA ticket to re-examine the `Checkpoint` design, including proposing emphasis on typed objects and formal interfaces.

File: tests/checkpoint/test_checkpoint_with_fluent_datasources.py [hunk 3]
```diff
@@ -1196,15 +1150,15 @@ def test_newstyle_checkpoint_instantiates_and_produces_a_validation_result_when_
...
+        "datasource_name": "my_sqlite_datasource",
+        "data_asset_name": "table_partitioned_by_date_column__A_query_asset_limit_5",
+    }
 
-    batch_request_1: FluentBatchRequest = FluentBatchRequest(
-        datasource_name="my_sqlite_datasource",
-        data_asset_name="table_partitioned_by_date_column__A_query_asset_limit_10",
-    )
+    batch_request_1 = {
```
Comment (alexsherstinsky) [inline tests/checkpoint/test_checkpoint_with_fluent_datasources.py:1160]: This is the correct change, due to the type requirements in the `Checkpoint` configuration.  There is a JIRA ticket to re-examine the `Checkpoint` design, including proposing emphasis on typed objects and formal interfaces.

File: tests/checkpoint/test_checkpoint_with_fluent_datasources.py [hunk 4]
```diff
@@ -1537,10 +1491,10 @@ def test_newstyle_checkpoint_instantiates_and_produces_a_validation_result_when_
     # create expectation suite
     context.add_expectation_suite("my_expectation_suite")
 
-    batch_request: FluentBatchRequest = FluentBatchRequest(
-        datasource_name="my_sqlite_datasource",
-        data_asset_name="table_partitioned_by_date_column__A_query_asset_limit_5",
-    )
+    batch_request = {
```
Comment (alexsherstinsky) [inline tests/checkpoint/test_checkpoint_with_fluent_datasources.py:1496]: This is the correct change, due to the type requirements in the `Checkpoint` configuration.  There is a JIRA ticket to re-examine the `Checkpoint` design, including proposing emphasis on typed objects and formal interfaces.

File: tests/checkpoint/test_checkpoint_with_fluent_datasources.py [hunk 5]
```diff
@@ -2314,10 +2268,10 @@ def test_newstyle_checkpoint_instantiates_and_produces_a_validation_result_when_
...
```
Comment (alexsherstinsky) [inline tests/checkpoint/test_checkpoint_with_fluent_datasources.py:2273]: This is the correct change, due to the type requirements in the `Checkpoint` configuration.  There is a JIRA ticket to re-examine the `Checkpoint` design, including proposing emphasis on typed objects and formal interfaces.

File: tests/checkpoint/test_simple_checkpoint.py [hunk 6]
```diff
@@ -129,9 +83,7 @@ def two_validations(
 @pytest.mark.integration
 def test_simple_checkpoint_default_properties_with_no_optional_arguments(
     empty_data_context,
-    store_validation_result_action,
-    store_eval_parameter_action,
-    update_data_docs_action,
+    common_action_list,
```
Comment (Shinnnyshinshin) [inline tests/checkpoint/test_simple_checkpoint.py:86]: :rocket:

File: tests/conftest.py [hunk 7]
```diff
@@ -1701,6 +1702,376 @@ def titanic_data_context_with_fluent_pandas_and_sqlite_datasources_with_checkpoi
     return context
 
 
+@pytest.fixture
+def titanic_data_context_with_fluent_pandas_datasources_stats_enabled_with_checkpoints_v1_with_templates(
```
Comment (Shinnnyshinshin) [inline tests/conftest.py:1706]: So I have a larger question around the scope of the `titanic_data_context_with_fluent_pandas_datasources_with_checkpoints_v1_with_empty_store_stats_enabled` fixture, stemming from issues we have been seeing around tests having "shared state". Here and in other places, we see a "starter" DataContext (`titanic_data_context_with_fluent_pandas_datasources_with_checkpoints_v1_with_empty_store_stats_enabled` in this case), and having additional datasources, and configs added -> sent to the test. 

I guess my question is : "are we sure that our "starter" DataContext is function-scoped in our tests (which would be the default)?

**Review by Shinnnyshinshin — APPROVED** [review]
Approving with a non-blocking question

File: tests/conftest.py (hunk 7 above)
Comment (alexsherstinsky) [reply tests/conftest.py:1706]: @Shinnnyshinshin We can try making it function-scoped, although other similar start `DataContext` fixtures are not, and I would prefer that we can conduct this experiment as a separate effort (for what it is worth, in my local tests, I have not run into any problems).  Thank you.

**Pull request merged**